lib/cdo/aws/dms.rb

```diff
@@ -65,5 +67,131 @@ def self.tasks(file)
         [task_name, {rules: rules}]
       end
     end
+
+    class ReplicationTask
+      # Get list of Replication Tasks that should be run on a specified schedule.
+      # @param schedule [String] Frequency that task should be executed ('daily', 'weekly', etc.)
+      # Returns array of Cdo::DMS::ReplicationTask
+      def self.tasks_by_frequency(schedule)
+        dms_client = Aws::DatabaseMigrationService::Client.new
+        replication_tasks = dms_client.describe_replication_tasks({without_settings: true}).replication_tasks
+        aws_tasks = replication_tasks.select do |task|
+          dms_client.
+            list_tags_for_resource({resource_arn: task.replication_task_arn}).
+            tag_list.
+            any? {|tag| tag.key == 'schedule' && tag.value == schedule}
+        end
+        aws_tasks.map do |aws_task|
+          Cdo::DMS::ReplicationTask.new(aws_task.replication_task_arn)
```

Comment:
(small/optional) you could pass in `dms_client` as the second parameter here, to reuse the same client across all tasks.

Reply:
Done!